Library/Homebrew/upgrade.rb

```diff
@@ -276,6 +276,20 @@ def check_installed_dependents(
         installed_formulae.flat_map(&:runtime_installed_formula_dependents)
                           .uniq
                           .select(&:outdated?)
+
+      # Ensure we never attempt a source build for outdated dependents of upgraded formulae.
+      outdated_dependents, skipped_dependents = outdated_dependents.partition do |dependent|
+        dependent.bottled? && dependent.deps.all?(&:bottled?)
+      end
+
+      if skipped_dependents.present?
+        opoo <<~EOS
+          The following dependents of upgraded formulae are outdated but will not
+          be upgraded:
```

Comment:
```suggestion
          The following dependents of upgraded formulae are outdated but will not
          be upgraded because they are not bottled:
```